src/features/applications/components/ApplicationForm.tsx

```diff
@@ -78,7 +77,16 @@ export function ApplicationForm() {
         persist="application-draft"
         schema={ApplicationCreateSchema}
         onSubmit={async ({ profile, application, applicationVerification }) => {
-          create.mutate({ application, applicationVerification, profile });
+          try {
+            const encryptedData = await ecrypt.mutateAsync(
+              applicationVerification,
+            );
+            application = { ...application, encryptedData };
+            create.mutate({ application, profile });
+          } catch (error) {
+            throw new Error("Failed to submit application!");
```

Comment:
Typo in ecrypt?

I think if using mutate instead of mutateAsync, the error will be available in encrypt.error if you want to render an error to the user. Might not be applicable here but good to know in the future.